sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala

```diff
@@ -393,6 +397,14 @@ object JavaTypeInference {
             inputObject :: Nil,
             returnNullable = false)
 
+        case c if c == classOf[java.sql.Date] && SQLConf.get.datetimeJava8ApiEnabled =>
```

Comment:
it's for dataset, so we shouldn't check the config here. If users want to collect a `Dataset[LocalDate]`, Spark must return `LocalDate` no matter what the config value is.

So here we should do
```
case c if c == classOf[LocalDate] =>
  StaticInvoke...
```

Reply:
fixed